controllers/clusteroperator_controller.go

```diff
@@ -180,6 +191,33 @@ func getResources(infra *configv1.Infrastructure) []client.Object {
 	return nil
 }
 
+func (r *CloudOperatorReconciler) composeConfig(ctx context.Context) (operatorConfig, error) {
```

Comment:
We should pass the infra in here rather than getting in this function, as we have it in the parent scope anyway

Reply:
It will later use more then just *v1.Infrastructure. The one in parent scope will eventually go away, and we still operate on cache here. So two separate calls in my opinion is better option, as we won't change the method signature much with it.

Reply:
This feels like premature optimisation to me. We should implement things as they are today, because while you say things will change in the future, inevitably they will probably not or things will change. 

Changing the signature of a private function is a non-issue, I still stand by passing it in here as it makes this function more isolated and easier to understand. 

I'm going to revise my earlier suggestion though, we actually only need the `platform`, we should pass that in from the parent scope as we have it there readily available, makes this function much smaller